tldraw/tldraw
# Snap to grid when creating shapes

TLD-2817

TLD-2816

This PR makes sure that shapes snap to the grid when created. It adds a ```maybeSnapToGrid``` function, which can be used to push a shape onto the grid if grid mode is enabled, both when click-creating and when drag-creating.

1. Any shapes using the basebox shape tool (i.e frames)
2. Geo shapes
3. Both arrow handles
4. Line shapes, including shift-clicking
5. Note shapes (when translating, note shapes prefer adjacent note positions over grid)
6. Text shapes
7. Aligns uploaded assets using the top left of the selection bounds.
8. Does not snap to the grid when snap indicators are being shown

It also adds tests for this behaviour

### Change type


### Test plan

1. Enable grid
9. Click-create a note shape off the grid
10. It should snap to the grid
11. Add an asset, it should align with the grid


### Release notes

- Shapes snap to grid on creation, or when adding points.
